src/network/networkd-manager.c

```diff
@@ -1204,6 +1204,52 @@ int manager_set_hostname(Manager *m, const char *hostname) {
         return 0;
 }
 
+static int set_static_hostname_handler(sd_bus_message *m, void *userdata, sd_bus_error *ret_error) {
+        const sd_bus_error *e;
+        int r;
+
+        assert(m);
+
+        e = sd_bus_message_get_error(m);
+        if (e) {
+                r = sd_bus_error_get_errno(e);
+                log_warning_errno(r, "Could not set static hostname: %s", bus_error_message(e, r));
+        }
+
+        return 1;
+}
+
+int manager_set_static_hostname(Manager *m, const char *hostname) {
```

Comment:
How about to merge this with `manager_set_hostname()`? They are almost equivalent.